--- mwbmatching.cpp	2021-12-20 15:54:29.765011826 -0500
+++ mwbmatching.cpp	2021-12-20 15:54:26.624135575 -0500
@@ -45,13 +45,16 @@
 
 int mwbmatching::check (graph& G) 
 {
-	if (!set_vars_executed)
+  if (!set_vars_executed)
     {
-		return(GTL_ERROR);
+      return(GTL_ERROR);
     }
-	if ((G.number_of_nodes() <= 1) || (!G.is_connected()) || (!G.is_directed()))
+  if ((G.number_of_nodes() <= 1) /*|| (!G.is_connected())*/ || (!G.is_directed()))
     {
-		return(GTL_ERROR);
+      if(!G.is_connected()) {
+        fprintf(stderr, "\nnot connected\n");
+      }
+      return(GTL_ERROR);
     }
     return GTL_OK;
 }
@@ -279,7 +282,7 @@
 	// Clean up
 	fh_free (pq);
 
-	
+        return 0;
 
 }
 
